contrib/kind.sh

```diff
@@ -205,307 +203,339 @@ print_params()
-cat ${KUBECONFIG}
+build_ovn_image() {
+  if [ "$OVN_IMAGE" == local ]; then
+    # Build ovn docker image
+    pushd ../go-controller
+    make
+    popd
+
+    # Build ovn kube image
+    pushd ../dist/images
+    # Find all built executables, but ignore the 'windows' directory if it exists
+    BINS=$(find ../../go-controller/_output/go/bin/ -maxdepth 1 -type f | xargs)
+    sudo cp -f ${BINS} .
+    echo "ref: $(git rev-parse  --symbolic-full-name HEAD)  commit: $(git rev-parse  HEAD)" > git_info
+    docker build -t ovn-daemonset-f:dev -f Dockerfile.fedora .
+    OVN_IMAGE=ovn-daemonset-f:dev
+    popd
+  fi
+}
 
-# Docker disables IPv6 globally inside containers except in the eth0 interface.
-# Kind enables IPv6 globally the containers ONLY for dual-stack and IPv6 deployments.
-# Ovnkube-node tries to move all global addresses from the gateway interface to the
-# bridge interface it creates. This breaks on KIND with IPv4 only deployments, because the new
-# internal bridge has IPv6 disable and can't move the IPv6 from the eth0 interface.
-# We can enable IPv6 always in the container, since the docker setup with IPv4 only
-# is not very common. 
-KIND_NODES=$(kind get nodes --name ${KIND_CLUSTER_NAME})
-for n in $KIND_NODES; do
-  docker exec $n sysctl net.ipv6.conf.all.disable_ipv6=0
-  docker exec $n sysctl net.ipv6.conf.all.forwarding=1
-done
+create_ovn_kube_manifests() {
+  # Detect API IP address for OVN
 
-if [ "${GITHUB_ACTIONS:-false}" == "true" ]; then
-  # Patch CoreDNS to work in Github CI
-  # 1. Github CI doesn´t offer IPv6 connectivity, so CoreDNS should be configured
-  # to work in an offline environment:
-  # https://github.com/coredns/coredns/issues/2494#issuecomment-457215452
-  # 2. Github CI adds following domains to resolv.conf search field:
-  # .net.
-  # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
-  # otherwise pods stops trying to resolve the domain.
-  # Get the current config
-  original_coredns=$(kubectl get -oyaml -n=kube-system configmap/coredns)
-  echo "Original CoreDNS config:"
-  echo "${original_coredns}"
-  # Patch it
-  fixed_coredns=$(
-    printf '%s' "${original_coredns}" | sed \
-      -e 's/^.*kubernetes cluster\.local/& net/' \
-      -e '/^.*upstream$/d' \
-      -e '/^.*fallthrough.*$/d' \
-      -e '/^.*forward . \/etc\/resolv.conf$/d' \
-      -e '/^.*loop$/d' \
-  )
-  echo "Patched CoreDNS config:"
-  echo "${fixed_coredns}"
-  printf '%s' "${fixed_coredns}" | kubectl apply -f -
-fi
+  # Despite OVN run in pod they will only obtain the VIRTUAL apiserver address
+  # and since OVN has to provide the connectivity to service
+  # it can not be bootstrapped
 
-if [ "$OVN_IMAGE" == local ]; then
-  # Build ovn docker image
-  pushd ../go-controller
-  make
-  popd
+  # This is the address of the node with the control-plane
+  API_URL=$(kind get kubeconfig --internal --name ${KIND_CLUSTER_NAME} | grep server | awk '{ print $2 }')
```

Comment:
we can move this to `detect_apiserver_ip() {` and explain that the function returns 2 values, the internal apiserver IP in API_URL used for the in-cluster communication, and the external apiserver IP in API_IP used for the out of cluster communication

Reply:
ok, see new `detect_apiserver_ip`